File: backend/src/main/java/com/example/testproject/entity/Dataset.java
package com.example.testproject.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.UUID;

/**
 * 数据集实体 - 对应 datasets 表
 */
@Data
@TableName("datasets")
public class Dataset implements Serializable {
    
    private static final long serialVersionUID = 1L;
    
    /** 数据集ID */
    @TableId(type = IdType.ASSIGN_UUID)
    private UUID id;
    
    /** 数据集名称 */
    private String name;
    
    /** 描述 */
    private String description;
    
    /** 所有者ID */
    private UUID ownerId;
    
    /** 数据类型 (timeseries/raster/vector/netcdf/csv/geotiff/shapefile) */
    private String dataType;
    
    /** 存储类型 (s3/local/nfs) */
    private String storageType;
    
    /** 存储路径 */
    private String storagePath;
    
    /** 文件大小 (字节) */
    private Long fileSize;
    
    /** 校验和 */
    private String checksum;
    
    /** 元数据 (JSON) */
    private String metadata;
    
    /** 空间边界 (WKT格式) */
    private String spatialBounds;
    
    /** 时间范围开始 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime temporalStart;
    
    /** 时间范围结束 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime temporalEnd;
    
    /** 下载次数 */
    private Integer downloadCount;
    
    /** 可见性 (private/public/organization) */
    private String visibility;
    
    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;
    
    /** 更新时间 */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updatedAt;
    
    /** 删除时间 */
    @TableLogic
    private LocalDateTime deletedAt;
}

File: frontend/src/types/dataset.ts
export type DataType = 'timeseries' | 'raster' | 'vector' | 'netcdf' | 'csv' | 'geotiff' | 'shapefile';
export type StorageType = 's3' | 'local' | 'nfs';
export type DatasetVisibility = 'private' | 'public' | 'group';

export interface Dataset {
  id: string;
  name: string;
  description?: string;
  ownerId: string;
  dataType: DataType;
  storageType: StorageType;
  storagePath: string;
  fileSize?: number;
  checksum?: string;
  metadata?: string;
  spatialBounds?: string;
  temporalStart?: string;
  temporalEnd?: string;
  downloadCount: number;
  visibility: DatasetVisibility;
  createdAt: string;
  updatedAt: string;
}

export const DATASET_DATA_TYPE_OPTIONS: { value: DataType; label: string; color: string }[] = [
  { value: 'timeseries', label: '时间序列', color: 'blue' },
  { value: 'raster', label: '栅格数据', color: 'green' },
  { value: 'vector', label: '矢量数据', color: 'orange' },
  { value: 'netcdf', label: 'NetCDF', color: 'purple' },
  { value: 'csv', label: 'CSV', color: 'cyan' },
  { value: 'geotiff', label: 'GeoTIFF', color: 'geekblue' },
  { value: 'shapefile', label: 'Shapefile', color: 'magenta' },
];

export const DATASET_DATA_TYPE_MAP: Record<DataType, { label: string; color: string }> = {
  timeseries: { label: '时间序列', color: 'blue' },
  raster: { label: '栅格数据', color: 'green' },
  vector: { label: '矢量数据', color: 'orange' },
  netcdf: { label: 'NetCDF', color: 'purple' },
  csv: { label: 'CSV', color: 'cyan' },
  geotiff: { label: 'GeoTIFF', color: 'geekblue' },
  shapefile: { label: 'Shapefile', color: 'magenta' },
};

export const DATASET_VISIBILITY_OPTIONS: { value: DatasetVisibility; label: string; color: string }[] = [
  { value: 'private', label: '私有', color: 'default' },
  { value: 'public', label: '公开', color: 'success' },
  { value: 'group', label: '组内', color: 'warning' },
];

export const DATASET_VISIBILITY_MAP: Record<DatasetVisibility, { label: string; color: string }> = {
  private: { label: '私有', color: 'default' },
  public: { label: '公开', color: 'success' },
  group: { label: '组内', color: 'warning' },
};

export interface DatasetPreviewResponse {
  headers: string[];
  rows: Record<string, unknown>[];
  totalRows: number;
  previewRows: number;
}

// 文件扩展名到数据类型的映射
export const FILE_EXTENSION_DATA_TYPE_MAP: Record<string, DataType> = {
  '.csv': 'csv',
  '.nc': 'netcdf',
  '.nc4': 'netcdf',
  '.tif': 'geotiff',
  '.tiff': 'geotiff',
  '.shp': 'shapefile',
  '.ts': 'timeseries',
  '.asc': 'raster',
};
